### PR TITLE
Add ANGLE Renderer | Fix Enivron not being set

### PR DESCRIPTION
**Pull request by thejudge156**
Exactly what the title says. 

Using the old ANGLE branch and some changes in Pojav and ANGLE, we have a working version of the ANGLE renderer in Pojav! 

Tested Multiplayer, and I will test Sodium and Iris in the coming days, unless anyone else wants to build this themselves and test for me, of course. 😀 

[ANGLE Repo](https://github.com/thejudge156/angle/tree/origin/new-main)

### CODE REVIEW

**Comment by thejudge156** [comment]
Oh, I am currently waiting for my action to finish and I will upload the rest of the arch libs.

**Comment by thejudge156** [comment]
ARM-64 is the only arch supported until the build is done in an hour.

File: app_pojavlauncher/src/main/java/net/kdt/pojavlaunch/utils/JREUtils.java [hunk 1]
```diff
@@ -222,7 +222,9 @@ public static void setJavaEnvironment(Activity activity) throws Throwable {
         envMap.put("REGAL_GL_VERSION", "4.5");
         if(LOCAL_RENDERER != null) {
             envMap.put("POJAV_RENDERER", LOCAL_RENDERER);
+            envMap.put("LIBGL_ES", Character.toString(LOCAL_RENDERER.charAt(8)));
```
Comment (Mathias-Boulay) [inline app_pojavlauncher/src/main/java/net/kdt/pojavlaunch/utils/JREUtils.java]: The LIBGL_ES version is already set by ExtraCore a few lines prior. This should be removed as far as I'm aware

File: app_pojavlauncher/src/main/jni/egl_bridge.c [hunk 2]
```diff
@@ -615,38 +615,38 @@ void pojav_openGLOnUnload() {
...
         } break;
     }
 }
 
 JNIEXPORT void JNICALL Java_net_kdt_pojavlaunch_utils_JREUtils_setupBridgeWindow(JNIEnv* env, jclass clazz, jobject surface) {
-    pojav_environ->pojavWindow = ANativeWindow_fromSurface(env, surface);
-    if(pojav_environ->config_renderer == RENDERER_GL4ES) {
-        gl_setup_window();
-    }
+pojav_environ->pojavWindow = ANativeWindow_fromSurface(env, surface);
```
Comment (Mathias-Boulay) [inline app_pojavlauncher/src/main/jni/egl_bridge.c]: For some reason, you broke a lot of indentations in this file

File: app_pojavlauncher/src/main/assets/components/lwjgl3/version [hunk 3]
```diff
@@ -1 +1 @@
-1664111720511
\ No newline at end of file
+1665886806915
```
Comment (Mathias-Boulay) [inline app_pojavlauncher/src/main/assets/components/lwjgl3/version:1]: no need for this file as well

File: app_pojavlauncher/src/main/res/values/headings_array.xml [hunk 4]
```diff
@@ -4,6 +4,7 @@
         <item name="2">@string/mcl_setting_renderer_gles2_4</item>
         <!-- <item name="4">@string/mcl_setting_renderer_vulkan_zink</item> -->
         <item name="5">@string/mcl_setting_renderer_virgl</item>
+        <item name="6">TINYWRAPPER ANGLE VULKAN</item>
```
Comment (Mathias-Boulay) [inline app_pojavlauncher/src/main/res/values/headings_array.xml]: hardcoded string, that's a no no

**Review by khanhduytran0 — COMMENTED** [review]
`gl_bridge.c`
Take `const ` out then it can be way more simplified to
```c
if(strncmp(getenv("POJAV_RENDERER"), "opengles3_desktopgl", 19) == 0) {
  egl_attributes[13] = EGL_OPENGL_BIT;
} else {
  egl_attributes[13] = EGL_OPENGL_ES2_BIT;
}
```

File: app_pojavlauncher/src/main/java/net/kdt/pojavlaunch/utils/JREUtils.java (hunk 1 above)
Comment (thejudge156) [reply app_pojavlauncher/src/main/java/net/kdt/pojavlaunch/utils/JREUtils.java]: I can wrap it in a if statement, it is used for ANGLE context creation and from what I am seeing it is going to stay as 1 or 2. But we should utilize 3 for ANGLE.

File: app_pojavlauncher/src/main/assets/components/lwjgl3/version (hunk 3 above)
Comment (thejudge156) [reply app_pojavlauncher/src/main/assets/components/lwjgl3/version:1]: This was a mess-up on my part.

**Comment by Abizhami** [comment]
Ixi

**Comment by AverageFemale** [comment]
I noticed when using this, the mouse seems to be off a bit from where the mouse is actually.

**Comment by AverageFemale** [comment]
> I noticed when using this, the mouse seems to be off a bit from where the mouse is actually.

Update on this, seems like after I restarted my phone the issue disappeared.. I have no clue what's happening, but it's never happened before.

**Comment by thejudge156** [comment]
Updates on when this will be merged in?

**Comment by thejudge156** [comment]
Anyone who wants to test this but needs a build, https://github.com/thejudge156/PojavLauncher/actions/runs/3342713282

**Comment by WesleyVanNeck** [comment]
I noticed with sodium that the world doesn't render

**Comment by GaucovyAero** [comment]
> I noticed with sodium that the world doesn't render

should be already fixed in the actions build on judges repo

**Comment by WesleyVanNeck** [comment]
> > I noticed with sodium that the world doesn't render
> 
> should be already fixed in the actions build on judges repo

well am using the latest action build

**Pull request merged**